AndroidOnChromeOSDragAndDropDemo/app/src/main/java/dev/hadrosaur/draganddropsample/MainActivity.kt

```diff
@@ -0,0 +1,172 @@
+                ACTION_DRAG_ENDED -> {
+                    // Back to default colour when drag is not in process
+                    view.setBackgroundColor(getColor(android.R.color.holo_purple))
+                    true
+                }
+                ACTION_DROP -> {
+                    requestDragAndDropPermissions(event) //Allow items from other applications
+                    val item = event.clipData.getItemAt(0)
+
+                    if (event.clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)) {
+                        if (item.text != null) {
+                            // Plain text item, show it on screen
+                            binding.textDropTarget.setTextSize(COMPLEX_UNIT_SP, 22f)
+                            binding.textDropTarget.text = item.text
+                        } else {
+                            // This is a plain text file. Read some characters and show them
+                            // Use ContentResolver to resolve the URI
+                            val contentUri = item.uri
+                            var parcelFileDescriptor: ParcelFileDescriptor? = null
+                            try {
+                                parcelFileDescriptor =
+                                    contentResolver.openFileDescriptor(contentUri, "r")
+                            } catch (e: FileNotFoundException) {
+                                e.printStackTrace()
+                                logd("File not found.")
+                                return false
+                            }
+
+                            if (parcelFileDescriptor == null) {
+                                logd("Error: could not load file: ${contentUri.toString()}")
+                                binding.textDropTarget.text = "Error: could not load file: ${contentUri.toString()}"
+                            } else {
+                                // Got the file descriptor, now read the file
+                                val fileDescriptor = parcelFileDescriptor.fileDescriptor
+                                val MAX_LENGTH = 5000
+                                val bytes = ByteArray(MAX_LENGTH)
+
+                                // Read the first MAX_LENGTH bytes of the file
+                                try {
+                                    val `in` = FileInputStream(fileDescriptor)
+                                    try {
+                                        `in`.read(bytes, 0, MAX_LENGTH)
+                                    } finally {
+                                        `in`.close()
+                                    }
+                                } catch (e: Exception) {
+                                    logd("Unable to read file: ${e.message}")
+                                }
+
+                                // Show CHARS_TO_READ chars in the UI
+                                val contents = String(bytes)
+                                val CHARS_TO_READ = 200
```

Comment:
Why are we reading 5000 bytes but limiting the string to 200? Seems that if we're supporting UTF-8 we can simply have `MAX_BYTES = CHARS_TO_READ * 4`?